src/dc_mimeparser.rs

```diff
@@ -1369,6 +1369,33 @@ mod tests {
     use proptest::prelude::*;
     use std::ffi::CStr;
 
+    #[test]
+    fn test_mimeparser_parse_with_gmime_testsuite() {
```

Comment:
To gain real confidence into our mime parsing capabilities, this should test mmime parsing directly and examine the structure of the output to be correct.